### DISCUSSION TITLE
Why does the extension requires permission for my browsing history?

### DISCUSSION THREAD

**Discussion by bheeshmpita** [crimx/ext-saladict#1918] [Q&A] [answered]
Secondly, what does the “Allow this extension to read and change all your data on websites that you visit" have to do with my data?

**Comment by crimx** [comment, marked as answer]
Saladict needs to read the selected elements and inject the dictionary panel to the page you are visiting. Hence the "read and change all your data on websites that you visit".

**Reply by bheeshmpita** [reply]
Thanks for the clarification. Does the user's data (browsing history except the word to be defined/translated) can be accessed by 3rd party or developer himself?

**Reply by crimx** [reply]
Yes if the developer chooses to collect and upload the data. Saladict does not collect browsing history or any other user private data.

**Reply by crimx** [reply]
Note that if the developer attempts to collect data aggressively, it is likely to be noticed by the extension store reviewer and the community (especially for large community).
